certbot-nginx/certbot_nginx/nginxparser.py

```diff
@@ -1,23 +1,29 @@
 """Very low-level nginx config parser based on pyparsing."""
+import copy
+import logging
 import string
 
 from pyparsing import (
-    Literal, White, Word, alphanums, CharsNotIn, Forward, Group,
+    Literal, White, Word, alphanums, CharsNotIn, Combine, Forward, Group,
     Optional, OneOrMore, Regex, ZeroOrMore)
 from pyparsing import stringEnd
 from pyparsing import restOfLine
 
+logger = logging.getLogger(__name__)
 
 class RawNginxParser(object):
     # pylint: disable=expression-not-assigned
     """A class that parses nginx configuration with pyparsing."""
 
     # constants
+    space = Optional(White())
```

Comment:
Is there some way that we can check that this grammar is equivalent to the grammar used by the real configuration parser inside of nginx itself? (I know that this grammar is currently passing all known test cases, which is certainly a good sign, but is there a practical way to compare code-against-code and say "apparently these grammars define the same language"?)